modules/templates/templates/templates.test.tsx

```diff
@@ -0,0 +1,732 @@
+      expect(typeSelectElement).toHaveValue("text");
+      expect(
+        screen.queryByTestId("template-detail.edit-field-config"),
+      ).not.toBeInTheDocument();
+
+      await userEvent.selectOptions(typeSelectElement, "number");
+      expect(typeSelectElement).toHaveValue("number");
+      await waitFor(() => {
+        expect(
+          screen.queryByTestId("template-detail.edit-field-config"),
+        ).toBeInTheDocument();
+      });
+
+      await userEvent.selectOptions(typeSelectElement, "number");
+      expect(typeSelectElement).toHaveValue("number");
+      await waitFor(() => {
+        expect(
+          screen.queryByTestId("template-detail.edit-field-config"),
+        ).toBeInTheDocument();
+      });
+
+      await userEvent.selectOptions(typeSelectElement, "number");
+      expect(typeSelectElement).toHaveValue("number");
+      await waitFor(() => {
+        expect(
+          screen.queryByTestId("template-detail.edit-field-config"),
+        ).toBeInTheDocument();
+      });
+    });
+
+    it("Checks config modal opens when config button clicked", async () => {
+      render(<TemplateDetail />);
+
+      await waitFor(() => {
+        expect(screen.getByTestId("table-row-0"));
+      });
+
+      const templateEdit = screen.getByTestId("template-detail.add-field");
+      act(() => templateEdit.click());
+
+      await waitFor(() => {
+        expect(
+          screen.getByTestId("template-detail.field-type-select"),
+        ).toBeInTheDocument();
+      });
+
+      const typeSelectElement = screen.getByTestId(
+        "template-detail.field-type-select",
+      );
+
+      await userEvent.selectOptions(typeSelectElement, "number");
+      expect(typeSelectElement).toHaveValue("number");
+      await waitFor(() => {
+        expect(
+          screen.queryByTestId("template-detail.edit-field-config"),
+        ).toBeInTheDocument();
+      });
+
+      const configButton = screen.getByTestId(
+        "template-detail.edit-field-config",
+      );
+      act(() => configButton.click());
+
+      const modal = await screen.findByTestId("template-detail.field-modal");
+      expect(modal).toBeInTheDocument();
+    });
+
+    it("Checks select config modal is populated", async () => {
+      render(<TemplateDetail />);
+
+      await waitFor(() => {
+        expect(screen.getByTestId("table-row-0"));
+      });
+
+      const templateEdit = screen.getAllByTestId("template-detail.edit");
+      act(() => templateEdit[1].click());
+
+      await waitFor(() => {
+        expect(
+          screen.getByTestId("template-detail.field-type-select"),
+        ).toBeInTheDocument();
+      });
+
+      const typeSelectElement = screen.getByTestId(
+        "template-detail.field-type-select",
+      );
+      expect(typeSelectElement).toHaveValue("select");
+
+      await waitFor(() => {
+        expect(
+          screen.queryByTestId("template-detail.edit-field-config"),
+        ).toBeInTheDocument();
+      });
```

Comment:
We can avoid all this is fort this test cases the property is already a template, we can `mockResolveOnce` on the API to give us the desire status :D  (Same for number and multiselect)